tests/suites/test_suite_psa_crypto_se_driver_hal_mocks.function

```diff
@@ -0,0 +1,207 @@
+/* BEGIN_HEADER */
+#include "psa_crypto_helpers.h"
+#include "psa/crypto_se_driver.h"
+
+#include "psa_crypto_se.h"
+
+static struct
+{
+    uint16_t called;
+    psa_key_slot_number_t key_slot;
+    psa_key_attributes_t attributes;
+    size_t data_length;
+} mock_import_data;
+
+static struct
+{
+    uint16_t called;
+    psa_key_slot_number_t slot_number;
+    size_t data_size;
+} mock_export_data;
+
+static struct
+{
+    uint16_t called;
+} mock_allocate_data;
+
+static struct
+{
+    uint16_t called;
+    psa_key_slot_number_t slot_number;
+} mock_destroy_data;
+
+static void mock_teardown( void )
+{
+    memset( &mock_import_data, 0, sizeof( mock_import_data ) );
+    memset( &mock_export_data, 0, sizeof( mock_export_data ) );
+    memset( &mock_allocate_data, 0, sizeof( mock_allocate_data ) );
+    memset( &mock_destroy_data, 0, sizeof( mock_destroy_data ) );
+}
+
+static psa_status_t mock_import( psa_drv_se_context_t *drv_context,
+                                 psa_key_slot_number_t key_slot,
+                                 const psa_key_attributes_t *attributes,
+                                 const uint8_t *data,
+                                 size_t data_length,
+                                 size_t *bits )
+{
+    (void) drv_context;
+    (void) data;
+    (void) bits;
+
+    mock_import_data.called++;
+    mock_import_data.key_slot = key_slot;
+    mock_import_data.attributes.core.lifetime = attributes->core.lifetime;
+    mock_import_data.attributes.core.type = attributes->core.type;
```

Comment:
Why do we need to save the attributes at all? I know we have asserts later on, but that's only testing that the mock saved this data properly, and if the mock doesn't save the data, we wouldn't need a test either. What in the core implementation is being tested if anything by having the mock attributes set here?

Perhaps we are testing that the core will pass the attributes down to the driver properly, in which case this makes sense. However, we should be checking more than just type and lifetime, shouldn't we?

Reply:
I've updated this to check all the set attributes.